Update compiling_workflows.rst (#5009)
PennyLaneAI/pennylane: doc/introduction/compiling_workflows.rst

    @qml.qjit
    @qml.qnode(dev)
    def circuit(params):
        qml.Hadamard(0)
        qml.RX(jnp.sin(params[0]) ** 2, wires=1)
        qml.CRY(params[0], wires=[0, 1])
        qml.RX(jnp.sqrt(params[1]), wires=1)
        return qml.expval(qml.PauliZ(1))

The :func:`~.qjit` decorator can also be used on hybrid functions --
that is, functions that include both QNodes and classical processing.

.. code-block:: python

    @qml.qjit
    def hybrid_function(params, x):
        grad = qml.grad(circuit)(params)[0]
        return jnp.abs(grad - x) ** 2

In addition, functions that are compiled with ``@jax.jit`` can contain calls
to qjit-compiled functions. For example, below we compile the full optimization loop,

> using ``@jax.jit``, for example when training models:

Reviewer: @josh146 line 79, please change `cost` to `circuit`?

Reviewer: It's really bothering that you can add any comments to that line in GH

Author: done!

> In addition, functions that are compiled with ``@jax.jit`` can contain calls
> to qjit-compiled functions. For example, below we compile the full optimization loop,
> using ``@jax.jit``, for example when training models:

Author: ```suggestion
In addition, functions that are compiled with ``@jax.jit`` can contain calls
to qjit-compiled functions. For example, below we compile a full optimization loop,
using ``@jax.jit``:
```

    import jaxopt

>     @qml.qjit

Reviewer 2: Just confirming if the change was tested? :)

>     @qml.qjit

Reviewer: I don't think this is correct!

Reviewer 2: Why is that the case?

        return params

> Compiling the entire hybrid workflow using ``@qjit`` however will lead to better

Reviewer: ```suggestion
Compiling the entire hybrid workflow using ``@qml.qjit`` however will lead to better
```
